specification/resources/DeploymentStacks.Management/models-whatIf.tsp

```diff
@@ -0,0 +1,384 @@
+  @visibility(Lifecycle.Read, Lifecycle.Create)
+  location?: string;
+
+  /** Resource tags. */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-no-record" "This is the expected type"
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "Model is shared between resource group, subscription, and management group scope."
+  tags?: Record<string>;
+}
+
+alias DeploymentStackWhatIfNameParameter = ResourceNameParameter<
+  Resource = DeploymentStackWhatIfResult,
+  KeyName = "deploymentStackWhatIfName",
+  SegmentName = "deploymentStackWhatIf",
+  NamePattern = "^[-\\w\\._\\(\\)]+$"
+>;
+
+/**
+ * DeploymentStack WhatIfResult Properties
+ */
+@added(Versions.v2025_07_01)
+model DeploymentStackWhatIfResultProperties is DeploymentStackSharedProperties {
+    
+    /**
+     * The deployment stack id to use as the basis for comparison.
+     */
+    deploymentStackId: string;
+
+    /**
+     * The interval to persist the deployment stack what-if result in ISO 8601 format.
+     */
+    retentionInterval: duration;
+
+/*
+    // TODO (R) - This only captures result format (for now), which is implicitly possible by get vs. POST on the whatif resource - so I'm thinking we skip this.
+    whatIfSettings?: DeploymentStackWhatIfSettings;
+*/
+    /**
+     * All of the changes predicted by the deployment stack what-if operation.
+     */
+    @visibility(Lifecycle.Read)
+    changes?: DeploymentStackWhatIfChange;
+
+    /**
+     * List of resource diagnostics detected by What-If operation.
+     */
+    @visibility(Lifecycle.Read)
+    @OpenAPI.extension("x-ms-identifiers", #[])
+    diagnostics?: DeploymentStackDiagnostics[];
+}
+
+/**
+ * Changes predicted to the deployment stack as a result of the what-if operation.
+ */
+@added(Versions.v2025_07_01)
+model DeploymentStackWhatIfChange { 
+    
+    /**
+     * List of resource changes predicted by What-If operation.
+     */
+    resourceChanges: DeploymentStackWhatIfResourceChange[];
+
+    /**
+     * Predicted changes to the deployment stack deny settings.
+     */
+    denySettingsChange: DeploymentStackChangeDelta<DenySettings>;
+
+    /**
+     * Predicted changes to the deployment scope for the deployment stack.
+     */
+    deploymentScopeChange: DeploymentStackChangeBase<string>;
+}
+
+/**
+ * Information about a single resource change predicted by What-If operation.
+ */
+@added(Versions.v2025_07_01)
+model DeploymentStackWhatIfResourceChange is ResourceChangeDelta {
+    /**
+     * The resource id of the resource being changed.
+     */
+    resourceId: string;
+
+    /**
+     * The resource id of the Deployment responsible for this change.
+     */
+    deploymentId: string;
+
+    /**
+     * The symbolic name of the resource being changed.
+     */
+    symbolicName: string;
+
+    /** 
+     * The extensible resource identifiers.
+     */
+    identifiers: unknown;    
```

Comment:
@kalbert312 can you provide some clarity on this? This property exists in deployments what-if, but I'm not sure if there should be a corresponding property on the Deployment stack (I didn't see it in master, at least).  Also wasn't sure if unknown made the most sense here.  If so, this will need a suppression tag.

Reply:
I must have made a mistake. It should be what https://github.com/Azure/bicep-extensibility/blob/main/docs/vnext/openapi.yaml specifies, which in TSP is `Record<unknown>`. I will update ResourceReference with this.

Reply:
I think it will also require an equivalent to `@@alternateType(DeploymentStackTemplateDefinition.template, unknown, "go")` which is what the Go breaking change reviewers added in `client.tsp`